packages/mdx/index.js

```diff
@@ -103,7 +104,13 @@ function sync(mdx, options) {
   const opts = Object.assign({}, DEFAULT_OPTIONS, options)
   const compiler = createCompiler(opts)
 
-  const { contents } = compiler.processSync(mdx)
+  const fileOpts = { contents: mdx };
+  if (opts.filename) {
```

Comment:
Is there ever no filename (or `resourcePath` in the loader)?

Reply:
I was expecting that `mdx` is not always called from the webpack loader, so I wanted to make the `filename` optional, a bit like [`babel-core` does](https://www.npmjs.com/package/babel-core#options).

Reply:
totes, thought we were still in the loader!